### PR TITLE
Various bug fixes

### LINKED ISSUE #3918: (WEB REPORT BY: amidaychi REMOTE: 198.245.63.50:7777) 2 nuke disk related issues
>Revision d2b95c949a53c4194728882075579ce1d2984f39 

>General description of the issue
1. Nuke disks can go in garbage bags if the nuke disk is on the same tile as something valid you pick up. 
2. combining mass driver with nuke disk z-level prevention causes player to get stuck at very edge of level.

>What you expected to happen
Garbage bag not to pick up disk, and to get bounced back in the direction of the station on zlevel attempted transition.

>What actually happened
Janitorborg picks up nuke disk in garbage bag during nuke ops round, stores garbage, bag, shoots self out of mass driver and is unreachable due to space buffer.

>Steps to reproduce if possible
Use garbage bag on valid baggable object on same tile as nuke disk to pick both up.

Shoot self out of mass driver afterwards, hug the blank nothingness till crew attritions the nuke ops away.

### PR DESCRIPTION
**Pull request by BarneyGumball**
Also fixes #3918
Fixes the bag part at least.  The "getting stuck on the edge of the z-level" part has already been fixed.

Other stuff:
- Fixed AI disable wire for Air Alarms having reversed functionality (connecting the wire would disable ai access and vice versa)
- Added `ignore_self` arg for `visible_message` to have the option for messages to not be displayed to the mob making the message

### CODE REVIEW

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -776,6 +776,8 @@
 				O.show_message(message, m_type)
 		else if (m_type & 2)
 			for (var/mob/O in hearers(src.loc, null))
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/human/emote.dm]: Shouldn't this be done via visible message instead?

Comment (BarneyGumball) [reply code/modules/mob/living/carbon/human/emote.dm]: I'm not sure what you mean.

File: code/game/machinery/alarm.dm [hunk 2]
```diff
@@ -635,17 +635,12 @@
 		return
 
 	if ( (get_dist(src, user) > 1 ))
```
Comment (clusterfack) [inline code/game/machinery/alarm.dm]: I'm just not sure how the following fixes the bug? Also to be honest in a fix of this bug I'd expect this get_dist to be changed to in_range, not to restrict you to that if the current fix makes sense.

Comment (BarneyGumball) [reply code/game/machinery/alarm.dm]: The distance check already happens somewhere else (in the ui I think) and it works as it should,  This was just overriding that check and forcing adjacency unnecessarily.

Tbh, I might just get rid of the rest of this if statement anyway and move the `aidisabled` check into `attack_ai`.

File: code/game/machinery/alarm.dm [hunk 3]
```diff
@@ -633,20 +639,9 @@
...
-
-		else if (istype(user, /mob/living/silicon) && aidisabled)
-			to_chat(user, "AI control for this Air Alarm interface has been disabled.")
-			user << browse(null, "window=air_alarm")
-			return
-
-	if(wiresexposed && (isMoMMI(user) || !istype(user, /mob/living/silicon)))
+	if(wiresexposed)
 		wires.Interact(user)
+		return
```
Comment (clusterfack) [inline code/game/machinery/alarm.dm]: I have this change in my other PR to fix a bug already, you fool we're going to conflict!

Comment (BarneyGumball) [reply code/game/machinery/alarm.dm:644]: No you did it for APCs.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 4]
```diff
@@ -771,12 +771,10 @@
...
 		INVOKE_EVENT(on_emote, list("mob"=src,"message"=message,"m_type"=m_type))
 
-		if (m_type & 1)
-			for (var/mob/O in viewers(src, null))
-				O.show_message(message, m_type)
-		else if (m_type & 2)
-			for (var/mob/O in hearers(src.loc, null))
-				O.show_message(message, m_type)
+		if (m_type & VISIBLE)
+			visible_message(message, null)
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/human/emote.dm]: Better to use named arguments than to pass null, particularly for this proc

**Comment by clusterfack** [comment]
Sorry if it seems like I'm nitpicking you on these fixes

**Comment by BarneyGumball** [comment]
Nah I'd rather you told me the right way to do things.  I'll never learn it if it isn't pointed out to me.

File: code/game/machinery/alarm.dm (hunk 3 above)
Comment (clusterfack) [inline code/game/machinery/alarm.dm]: Ah oh I see

**Comment by BarneyGumball** [comment]
I couldn't live with `message = src.message` so I changed the var name.  Triple checked `emote.dm` for mistakes and if any other files were passing named arguments to it.

**Comment by SonixApache** [comment]
Does putting the issuenumber in the commits even work for autoclosing?

**Comment by clusterfack** [comment]
Yes

**Pull request merged**